ci/tools/test_dcz.py

```
@@ -306,6 +313,27 @@ def vary_values(headers) -> str:
+
+    $zstd_bytes_out / $zstd_ratio are log-phase variables: nginx writes
+    the access_log line only after the response is fully sent, which can
+    race a fast local request. Poll instead of reading once, mirroring
+    test_encoding.py's validate_ratio_log.
+    """
+    deadline = time.time() + timeout
+    line = ""
+    while time.time() < deadline:
+        if path.exists():
+            content = path.read_text(encoding="utf-8", errors="replace")
+            lines = [l for l in content.splitlines() if l.strip()]
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)